### PR TITLE
AB#87828 Added fetch DeliveryPhase for milestone pages

### PR DESCRIPTION
**Pull request by Antek012**

### CODE REVIEW

File: src/HE.Investment.AHP.Domain/Delivery/Mappers/DeliveryPhaseEntityMapper.cs [hunk 1]
```diff
@@ -17,4 +20,29 @@ public static DeliveryPhaseDetails ToContract(IDeliveryPhaseEntity deliveryPhase
...
+            MapDate(deliveryPhase.StartOnSiteMilestone?.PaymentDate),
+            MapDate(deliveryPhase.CompletionMilestone?.CompletionDate),
+            MapDate(deliveryPhase.CompletionMilestone?.PaymentDate));
+    }
+
+    private static DateDetails? MapDate(DateValueObject? date)
+    {
+        return date != null
+            ? new DateDetails(
+                date.Date.Day.ToString(CultureInfo.InvariantCulture),
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/Delivery/Mappers/DeliveryPhaseEntityMapper.cs:43]: maybe Date property on DaveValueObject should be renamed to Value?

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/AcquisitionMilestone.cshtml [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+@using HE.Investment.AHP.Contract.Delivery
 @model HE.Investment.AHP.WWW.Models.Delivery.MilestoneViewModel
 @{
     var title = "Acquisition milestone";
     ViewData["Title"] = title;
 }
 
 <gds-div-grid-column-two-thirds>
-    <vc:back-link-with-callback url="@Url.Action("Index", "Home")"/>
+        <vc:workflow-back-button action="@DeliveryPhaseWorkflowState.AcquisitionMilestone.ToString()"/>
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.WWW/Views/DeliveryPhase/AcquisitionMilestone.cshtml:16]: wrong formatting

File: src/HE.Investments.Account.WWW/Controllers/OrganisationController.cs [hunk 3]
```diff
@@ -103,10 +103,10 @@ public async Task<IActionResult> CreateOrganisation(OrganisationDetailsViewModel
         return await this.ExecuteCommand(
             _mediator,
             command,
-            onSuccess: () => RedirectToAction(
+            onSuccess: async () => await Task.FromResult(RedirectToAction(
```
Comment (jjankowski87) [inline src/HE.Investments.Account.WWW/Controllers/OrganisationController.cs:106]: remove async/await keywords, those are not needed when returning Task.FromResult

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Delivery/Mappers/DeliveryPhaseEntityMapper.cs (hunk 1 above)
Comment (Antek012) [reply src/HE.Investment.AHP.Domain/Delivery/Mappers/DeliveryPhaseEntityMapper.cs]: changed

File: src/HE.Investments.Account.WWW/Controllers/OrganisationController.cs (hunk 3 above)
Comment (Antek012) [reply src/HE.Investments.Account.WWW/Controllers/OrganisationController.cs]: fixed

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/AcquisitionMilestone.cshtml (hunk 2 above)
Comment (Antek012) [reply src/HE.Investment.AHP.WWW/Views/DeliveryPhase/AcquisitionMilestone.cshtml]: fixed

File: src/HE.Investment.AHP.Domain/Delivery/ValueObjects/DateValueObject.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@ public DateValueObject(string? day, string? month, string? year, string fieldNam
         Build(day, month, year, fieldName, fieldLabel).CheckErrors();
     }
 
-    public DateTime Date { get; private set; }
+    public DateTime Value { get; private set; }
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Delivery/ValueObjects/DateValueObject.cs:13]: Use `DateOnly` class if possible

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Delivery/ValueObjects/DateValueObject.cs (hunk 4 above)
Comment (Antek012) [reply src/HE.Investment.AHP.Domain/Delivery/ValueObjects/DateValueObject.cs:13]: fix in next PR

**Pull request merged**